### ISSUE TITLE
Documentation incorrect for basic authentication

### ISSUE THREAD

**Issue by jknipp** [uesteibar/neuron#24]
Dependency versions:
* hackney: 1.11.0
* neuron: 0.9.1
* httpoison: 1.3.1

When following the example for using [basic authentication](https://github.com/uesteibar/neuron#usage), I'm receiving an error message from Hackney:

```
** (ArgumentError) argument error
    :erlang.list_to_binary([basic_auth: {"user", "paassword"}])
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney_bstr.erl:27: :hackney_bstr.to_binary/1
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney_headers_new.erl:193: anonymous fn/3 in :hackney_headers_new.to_iolist/1
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney_headers_new.erl:148: :hackney_headers_new.do_fold/3
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney_headers_new.erl:184: :hackney_headers_new.to_iolist/1
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney_request.erl:95: :hackney_request.perform/2
    (hackney) /Users/jaredknipp/dev/my_app/deps/hackney/src/hackney.erl:358: :hackney.send_request/2
    (httpoison) lib/httpoison/base.ex:633: HTTPoison.Base.request/9
    (neuron) lib/neuron.ex:98: Neuron.run/2
```

I believe `hackney` options should be specified by as the [`options` argument](https://github.com/edgurgel/httpoison/blob/master/lib/httpoison/base.ex#L631) via `connection_opts` to HTTPoison, not as headers. For example, using this for auth does work as expected:

```
Neuron.Config.set(connection_opts: [hackney: [basic_auth: {"user", "password"}]])
```

Optionally, you can set the `Authorization` header using the `headers` config option as so:

```
encoded_credentials = Base.encode64("user:password")
Neuron.Config.set(headers: ["Authorization": "Basic #{encoded_credentials}"])
```

If this can be confirmed, I can submit a PR w/ the updates.

**Comment by uesteibar** [comment]
hi @jknipp! Sorry for taking so long to reply, I thought I already did it 😭 

I recently cleaned up the Readme and I removed this example. The goal wasn't to provide the hackney example but to have an example on how to add headers. However, this is now moved to the hexdocs so this change would not be needed anymore.

Thanks for reporting! I messed up when writing the example and didn't notice until you opened this one :)
